Sources/ArgumentParser/Usage/UsageGenerator.swift

```diff
@@ -246,16 +246,35 @@ extension ErrorMessageGenerator {
   var invalidState: String {
     return "Internal error. Invalid state while parsing command line arguments."
   }
-  
+
+
   func unknownOptionMessage(origin: InputOrigin.Element, name: Name) -> String {
-    switch name {
-    case .long(let n):
-      return "Unknown option '--\(n)'"
-    case .short(let n):
-      return "Unknown option '-\(n)'"
-    case .longWithSingleDash(let n):
-      return "Unknown option '-\(n)'"
+    if case .short = name {
+      return "Unknown option '\(name.synopsisString)'"
+    }
+    
+    // An empirically derived magic number
+    let SIMILARITY_FLOOR = 4
```

Comment:
This was set at 7 because that is what [`git` does](https://github.com/git/git/blob/228f53135a4a41a37b6be8e4d6e2b6153db4a8ed/help.c#L592). However it leads us to suggest `--name` for `-bar` or `--verbose` here: https://github.com/apple/swift-argument-parser/pull/10/files#diff-f55d39a7e333981fb00eaa62155a2c8cL35-R39. The suggestions seemed to disimilar to allow.

Reply:
👍 7 did seem high!